lib/smb2/client.rb

```diff
@@ -74,6 +88,8 @@ def initialize(opts = {})
     @password    = opts.fetch(:password).encode("utf-8")
     @domain      = opts[:domain]
     @local_workstation = (opts[:local_workstation] || "")
+
```

Comment:
Since you're requiring required keyword argument support from Ruby 2.1, you can change these `opt.fetch` calls to required keyword arguments.  You can also change the `:domain` and `:local_workstation` to optional keyword arguments:

```ruby
  # @option opts [#send_packet,#recv_packet] :dispatcher
  # @option opts [String] :username UTF-8
  # @option opts [String] :password UTF-8
  # @option opts [String] :domain UTF-8 *(optional)*
  # @option opts [String] :local_workstation UTF-8 *(optional)*
  # @raise KeyError if required options are missing
  def initialize(dispatcher:, domain: nil, local_workstation: '', password:, username:)
    @dispatcher  = dispatcher
    @username    = username.encode("utf-8")
    @password    = password.encode("utf-8")
    @domain      = domain
    @local_workstation = local_workstation

    @tree_ids = []
  end
```

**NOTE: This change can be done in a separate Chore PR since it's not strictly related to adding Read files support**